test/controllers/drivers_controller_test.rb

```diff
@@ -2,103 +2,139 @@
       # Arrange
       # Ensure that there is at least one Driver saved
-
+      driver.save
       # Act
-
+      get drivers_path
+      
       # Assert
-
+      must_respond_with :success
     end
 
     it "responds with success when there are no drivers saved" do
       # Arrange
       # Ensure that there are zero drivers saved
-
+      drivers = Driver.all
+      expect(drivers.length).must_equal 0
       # Act
-
+      get drivers_path
       # Assert
-
+      must_respond_with :success
     end
   end
 
   describe "show" do
     it "responds with success when showing an existing valid driver" do
       # Arrange
       # Ensure that there is a driver saved
-
+      driver.save
       # Act
-
+      get driver_path(driver.id)
       # Assert
-
+      must_respond_with :success
     end
 
-    it "responds with 404 with an invalid driver id" do
+    it "responds with redirect with an invalid driver id" do
       # Arrange
       # Ensure that there is an id that points to no driver
 
       # Act
-
+      get driver_path(-1)
       # Assert
-
+      must_respond_with :redirect
     end
   end
 
   describe "new" do
     it "responds with success" do
+      # Act
+      get new_driver_path
+
+      # Assert
+      must_respond_with :success
     end
   end
 
   describe "create" do
     it "can create a new driver with valid information accurately, and redirect" do
       # Arrange
       # Set up the form data
+      driver_hash = {
+        driver: {
+          name: "Test Test",
+          vin: "12345678901234567",
+        },
+      }
 
       # Act-Assert
       # Ensure that there is a change of 1 in Driver.count
-
+      expect {
+        post drivers_path, params: driver_hash
+      }.must_change "Driver.count", 1
       # Assert
       # Find the newly created Driver, and check that all its attributes match what was given in the form data
       # Check that the controller redirected the user
-
+      new_driver = Driver.find_by(name: driver_hash[:driver][:name])
+      expect(new_driver.vin).must_equal driver_hash[:driver][:vin]
+      expect(new_driver.available).must_equal 'true'
+      
+      must_redirect_to driver_path(new_driver.id)
     end
 
-    it "does not create a driver if the form data violates Driver validations, and responds with a redirect" do
+    it "does not create a driver if the form data violates Driver validations, and responds with not_found" do
       # Note: This will not pass until ActiveRecord Validations lesson
       # Arrange
       # Set up the form data so that it violates Driver validations
-
+      driver_hash = {
+        driver: {
+          name: "Test",
+          vin: "",
+        },
+      }
       # Act-Assert
       # Ensure that there is no change in Driver.count
-
+      expect {
+        post drivers_path, params: driver_hash
+      }.must_differ "Driver.count", 0
       # Assert
       # Check that the controller redirects
-
+      must_respond_with :not_found
```

Comment:
Not found doesn't really make sense here.  Maybe use bad_request instead. 

```suggestion
      must_respond_with :bad_request
```